src/containers/WelcomePage/hooks/storedFilterHooks.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { useCallback, useEffect, useState } from "react";
+import { SingleValue } from "@ndla/select";
+import { fetchNode } from "../../../modules/nodes/nodeApi";
+import { Prefix } from "../components/TableComponent";
+
+type ReturnStateType<T> = [T, (v: T) => void];
+
+export const useStoredSubjectFilterHook = (
+  localStorageKey: string,
+  language: string,
+): ReturnStateType<SingleValue | undefined> => {
+  const [filterSubject, _setFilterSubject] = useState<SingleValue | undefined>(undefined);
+  const storedFilterSubject = localStorage.getItem(localStorageKey);
+
+  useEffect(() => {
+    if (storedFilterSubject) {
+      const updateFilterSubject = async () => {
+        const node = await fetchNode({
+          id: storedFilterSubject,
+          language,
+          taxonomyVersion: "default",
+        });
+        _setFilterSubject({ label: node.name, value: storedFilterSubject });
+      };
+      updateFilterSubject();
+    }
+  }, [language, storedFilterSubject]);
+
+  const setFilterSubject = useCallback(
+    (fs: SingleValue | undefined) => {
+      _setFilterSubject(fs);
+      fs ? localStorage.setItem(localStorageKey, fs.value) : localStorage.removeItem(localStorageKey);
+    },
+    [localStorageKey],
+  );
+
+  return [filterSubject, setFilterSubject];
+};
+
+const defaultPageSize = { label: "6", value: "6" };
+
+export const useStoredPageSizeHook = (localStorageKey: string): ReturnStateType<SingleValue> => {
+  const storedPageSize = localStorage.getItem(localStorageKey);
+
+  const [pageSize, _setPageSize] = useState<SingleValue>(
+    storedPageSize
+      ? {
+          label: storedPageSize,
+          value: storedPageSize,
+        }
+      : defaultPageSize,
+  );
+  const setPageSize = useCallback(
+    (p: SingleValue) => {
+      if (!p) return;
+      _setPageSize(p);
+      localStorage.setItem(localStorageKey, p.value);
+    },
+    [localStorageKey],
+  );
+
+  return [pageSize, setPageSize];
+};
+
+export const useStoredSortOptionHook = <T extends string>(
+  localStorageKey: string,
+  fallbackSortOption: Prefix<"-", T>,
+): ReturnStateType<Prefix<"-", T>> => {
+  const [sortOption, _setSortOption] = useState<Prefix<"-", T>>(
+    (localStorage.getItem(localStorageKey) as T) || fallbackSortOption,
+  );
+
+  const setSortOption = useCallback(
+    (s: Prefix<"-", T>) => {
+      _setSortOption(s);
+      localStorage.setItem(localStorageKey, s);
+    },
+    [localStorageKey],
+  );
+
+  return [sortOption, setSortOption];
+};
+export const useStoredToggleHook = (localStorageKey: string): ReturnStateType<boolean> => {
```

Comment:
Blæh, er dette egentlig en toggle? Føler jo egentlig på en måte at en toggle ikke skal ta i mot et parameter, men heller bare switche staten sin

Reply:
Åpen for andre forslag på navngivningen! Tanken var å bruke den for local storage på toggles, har du forslag til bedre navngivning på den?

Reply:
`useLocalStorageState` ellerno?

Reply:
Jaa, men denne er jo låst til at staten er boolean, så jeg syntes det og blir misvisende

Reply:
Isåfall har jeg ikke noen gode forslag, men det eksisterende er nok litt for generisk.

Reply:
`useStoredBooleanState`? `useLocalStorageBooleanState`? Ikke 10/10, men blir kanskje mer beskrivende

Reply:
Ville foreslått `useLocalStoredBooleans` eller `useLocalStoredBooleanStates`, alt ettersom om man mener at det ikke er beskrivende nok uten State.

Reply:
Da er navngivning oppdatert! Håper alle er fornøyde 😅 🤞

Reply:
Kan ikke se noe å utsette. Lar Jonas ha siste ord 👍